### PR TITLE
Prepare for putting documentation on readthedocs

### PR DESCRIPTION
**Pull request by bmerry**
- Pin all packages in doc-requirements.txt
- Regenerate boilerplate files with newer Sphinx (2.1)
- Remove documentation of the test modules
- Extract version info from katversion
- Fix a formatting issue in README.rst
- Add a .readthedocs.yml

### CODE REVIEW

**Comment by bmerry** [comment]
The build is failing because it needs ska-sa/katsdpjenkins#10.

**Comment by bmerry** [comment]
Draft version visible at https://katsdptelstate.readthedocs.io/en/latest/

File: doc-requirements.txt [hunk 1]
```diff
@@ -1,29 +1,31 @@
...
-pytz                        # via babel
-requests                    # via sphinx
-six                         # via packaging, sphinx
-snowballstemmer             # via sphinx
-sphinx-rtd-theme
-sphinx
-sphinxcontrib-websupport    # via sphinx
-typing; python_version<'3'  # via sphinx
-urllib3                     # via requests
+alabaster==0.7.12         # via sphinx
```
Comment (ludwigschwardt) [inline doc-requirements.txt]: I thought the reason we unpinned this is because they match the docker-base versions?

File: doc/conf.py [hunk 2]
```diff
@@ -1,170 +1,59 @@
...
+sys.path.insert(0, os.path.abspath('..'))
+import katsdptelstate
 
 
-# -- General configuration ------------------------------------------------
+# -- Project information -----------------------------------------------------
+
+project = 'katsdptelstate'
+copyright = '2019, South African Radio Astronomy Observatory'
+author = 'South African Radio Astronomy Observatory'
```
Comment (ludwigschwardt) [inline doc/conf.py]: katdal had "SARAO".

File: doc/conf.py [hunk 3]
```diff
@@ -1,170 +1,59 @@
...
+import sys
+sys.path.insert(0, os.path.abspath('..'))
+import katsdptelstate
 
 
-# -- General configuration ------------------------------------------------
+# -- Project information -----------------------------------------------------
+
+project = 'katsdptelstate'
+copyright = '2019, South African Radio Astronomy Observatory'
```
Comment (ludwigschwardt) [inline doc/conf.py]: katdal had "NRF".

File: doc/conf.py [hunk 4]
```diff
@@ -1,170 +1,59 @@
...
+author = 'South African Radio Astronomy Observatory'
+version = '.'.join(katsdptelstate.__version__.split('.')[:2])
```
Comment (ludwigschwardt) [inline doc/conf.py]: katdal had `__version__` for both.

File: doc-requirements.txt (hunk 1 above)
Comment (bmerry) [reply doc-requirements.txt]: Yes, but readthedocs doesn't know anything about docker-base, so we have to explicitly pin if we want readthedocs and Jenkins to build docs from the same versions.

File: doc/conf.py (hunk 3 above)
Comment (bmerry) [reply doc/conf.py]: I've updated katdal

File: doc/conf.py (hunk 4 above)
Comment (bmerry) [reply doc/conf.py]: I've updated katdal.

File: doc/conf.py (hunk 2 above)
Comment (bmerry) [reply doc/conf.py]: I've updated katdal.

**Pull request merged**